src/index.js

```diff
@@ -3,47 +3,49 @@ import Store from './modules/Store.js';
 
-document.addEventListener('DOMContentLoaded', UI.displayBooks);
+document.addEventListener('DOMContentLoaded', () => {
+  UI.displayBooks();
 
-document.querySelector('#book-form').addEventListener('submit', (e) => {
-  e.preventDefault();
-  const title = document.querySelector('#title').value;
-  if (title !== '') {
-    const book = new Book(title);
-    UI.addBookToList(book);
-    Store.addBook(book);
-    document.querySelector('#title').value = '';
-  }
-});
+  document.querySelector('#book-form').addEventListener('submit', (e) => {
+    e.preventDefault();
+    const title = document.querySelector('#title').value;
+    if (title !== '') {
+      const book = new Book(title);
+      UI.addBookToList(book);
+      Store.addBook(book);
+      document.querySelector('#title').value = '';
+    }
+  });
 
-document.querySelector('.book-list').addEventListener('click', (e) => {
-  if (e.target.classList.contains('delete')) {
-    UI.deleteBook(e.target);
-    const { isbn } = e.target.parentElement.parentElement.dataset;
-    Store.removeBook(isbn);
-  } else if (e.target.classList.contains('mybox')) {
-    const row = e.target.parentElement.parentElement;
-    row.classList.toggle('completed');
-    const { isbn } = row.dataset;
-    Store.toggleComplete(isbn);
-  }
-});
+  document.querySelector('.book-list').addEventListener('click', (e) => {
+    if (e.target.classList.contains('delete')) {
+      UI.deleteBook(e.target);
+      const { isbn } = e.target.parentElement.parentElement.dataset;
+      Store.removeBook(isbn);
+    } else if (e.target.classList.contains('mybox')) {
+      const row = e.target.parentElement.parentElement;
+      row.classList.toggle('completed');
+      const { isbn } = row.dataset;
+      Store.toggleComplete(isbn);
+    }
+  });
 
-document.querySelector('.book-list').addEventListener('input', (e) => {
-  if (e.target.classList.contains('Edit')) {
-    const { isbn } = e.target.parentElement.parentElement.dataset;
-    const newTitle = e.target.value;
-    UI.editBookTitle(isbn, newTitle);
-    Store.editBookTitle(isbn, newTitle);
-  }
-});
+  document.querySelector('.book-list').addEventListener('input', (e) => {
+    if (e.target.classList.contains('Edit')) {
+      const { isbn } = e.target.parentElement.parentElement.dataset;
+      const newTitle = e.target.value;
+      UI.editBookTitle(isbn, newTitle);
+      Store.editBookTitle(isbn, newTitle);
+    }
+  });
 
-document.querySelector('.ClearAll').addEventListener('click', () => {
-  UI.clearCompletedBooks();
-  Store.clearCompletedBooks();
-});
+  document.querySelector('.ClearAll').addEventListener('click', () => {
+    UI.clearCompletedBooks();
+    Store.clearCompletedBooks();
+  });
 
-document.querySelector('.ClearAll').addEventListener('click', () => {
-  UI.removeAllBooks();
-  Store.removeAllBooks();
+  document.querySelector('.ClearAll').addEventListener('click', () => {
+    UI.removeAllBooks();
+    Store.removeAllBooks();
+  });
```

Comment:
- [ ] The previously mentioned error still pertains where it says `list is null` and I believe the main problem here is that when you call the `UI.removeAllBooks();` function the container with a class name of `book-list` is already removed from the DOM and hence when you try to access it throws a `null`. So please carefully track the bug and make sure to fix it. ✍
- Here's the error once again:
  ![image](https://github.com/tchasinga/To-do-list-Microverse/assets/104968236/9c5864d5-37c1-444f-83a0-e4de46ecea7a)